src/App.tsx

```diff
@@ -16,16 +25,18 @@ export const App: React.FC = () => {
 
             <div className="control">
               <input
+                value={query}
                 type="text"
                 id="search-query"
                 className="input"
                 placeholder="Type search word"
+                onChange={(event) => setQuery(event.target.value)}
```

Comment:
you can optimize it and move function to variable;
So 
const handleChangeQuery = (event) => setQuery(event.target.value)
onChange={handleChangeQuery}